MonkeyBot/Modules/Polls/PollModule.cs

```diff
@@ -55,84 +33,86 @@ public PollModule(IPollService pollService)
 
-            var timeResponse = await _interactivity.WaitForMessageAsync(msg => msg.Author == ctx.Member && msg.ChannelId == ctx.Channel.Id, _timeOut);
+            var timeResponse =
+                await interactivity.WaitForMessageAsync(
+                    msg => msg.Author == ctx.Member && msg.ChannelId == ctx.Channel.Id,
+                    _timeOut);
             if (timeResponse.TimedOut)
             {
                 await ctx.ErrorAsync("You didn't respond in time. Please start over", "Timed out");
-                await ctx.Channel.DeleteMessagesAsync(new[] { setupMessage, pollMessage });
+                await ctx.Channel.DeleteMessagesAsync(new[] {setupMessageUpdater.Message, pollMessageUpdater.Message});
                 return;
             }
-            _timeParser ??= new Chronic.Parser(new Chronic.Options() { Context = Chronic.Pointer.Type.Future, EndianPrecedence = Chronic.EndianPrecedence.Little, FirstDayOfWeek = DayOfWeek.Monday });
-            Chronic.Span parsedTime = _timeParser.Parse(timeResponse.Result.Content);
+
+            var timeParser = new Chronic.Parser(new Chronic.Options()
+            {
+                Context = Chronic.Pointer.Type.Future,
+                EndianPrecedence = Chronic.EndianPrecedence.Little,
+                FirstDayOfWeek = DayOfWeek.Monday
+            });
+            Chronic.Span parsedTime = timeParser.Parse(timeResponse.Result.Content);
             if (parsedTime == null)
             {
                 await ctx.ErrorAsync("I couldn't understand this Date/Time. Please start over", "Invalid Time");
-                await ctx.Channel.DeleteMessagesAsync(new[] { setupMessage, pollMessage });
+                await ctx.Channel.DeleteMessagesAsync(new[] {setupMessageUpdater.Message, pollMessageUpdater.Message});
                 return;
             }
+
             DateTime endTime = parsedTime.ToTime();
             if (endTime < DateTime.Now)
             {
                 await ctx.ErrorAsync("The provided time is in the past. Please start over", "Invalid Time");
-                await ctx.Channel.DeleteMessagesAsync(new[] { setupMessage, pollMessage });
+                await ctx.Channel.DeleteMessagesAsync(new[] {setupMessageUpdater.Message, pollMessageUpdater.Message});
                 return;
             }
 
-            pollEmbed.WithFooter($"Poll will end on {endTime:dd.MM.yyyy} at {endTime:HH:mm \"UTC\"zz}");
-            pollMessage = await pollMessage.ModifyAsync(embed: pollEmbed.Build());
+            await pollMessageUpdater.SetEndTime(endTime);
 
             await ctx.Channel.DeleteMessageAsync(timeResponse.Result);
 
-            setupEmbed.WithDescription(introText + thirdInstruction);
-            await setupMessage.ModifyAsync(embed: setupEmbed.Build());
-            await setupMessage.CreateReactionAsync(_okEmoji);
+            await setupMessageUpdater.SetThirdInstruction();
+            await setupMessageUpdater.AddOkButton();
```

Comment:
If its not too much hassle it might be worth adding this button only after a minimum of 2 answers have been provided

Reply:
It can take some time, because moving creation of this btn requires changes how are those two interactions awaited in that while loop. Maybe would be better to create separate issue for that?

Reply:
Yes ok, lets leave that out for now